tests/integration/pox/stacking/direct-and-pooled-increase-2.spec.ts

```diff
@@ -0,0 +1,164 @@
+import { DevnetNetworkOrchestrator } from "@hirosystems/stacks-devnet-js";
+import { StacksTestnet } from "@stacks/network";
+import { uintCV } from "@stacks/transactions";
+import { Accounts } from "../../constants";
+import {
+  buildDevnetNetworkOrchestrator,
+  getNetworkIdFromEnv,
+  waitForStacksTransaction,
+} from "../../helpers";
+import {
+  expectAccountToBe,
+  getPoxInfo,
+  readRewardCyclePoxAddressListAtIndex,
+  waitForNextRewardPhase,
+} from "../helpers";
+import { broadcastStackSTX } from "../helpers-direct-stacking";
+import {
+  broadcastDelegateSTX,
+  broadcastDelegateStackIncrease,
+  broadcastStackAggregationCommitIndexed,
+} from "../helpers-pooled-stacking";
+
+describe("testing mixed direct and pooled stacking under epoch 2.1", () => {
+  let orchestrator: DevnetNetworkOrchestrator;
+  let timeline = {
+    epoch_2_0: 100,
+    epoch_2_05: 102,
+    epoch_2_1: 106,
+    pox_2_activation: 109,
+  };
+  const fee = 1000;
+
+  beforeAll(() => {
+    orchestrator = buildDevnetNetworkOrchestrator(getNetworkIdFromEnv());
+    orchestrator.start();
+  });
+
+  afterAll(() => {
+    orchestrator.terminate();
+  });
+
+  it("using direct stacking and delegate-stack-increase with auto-unlock,", async () => {
+    const network = new StacksTestnet({ url: orchestrator.getStacksNodeUrl() });
+
+    // Wait for Stacks genesis block
+    await orchestrator.waitForNextStacksBlock();
+    // Wait for block N+1 where N is the height of the next reward phase
+    await waitForNextRewardPhase(network, orchestrator, 1);
+
+    const blockHeight = timeline.pox_2_activation + 1;
+    const cycles = 1;
+
+    // Faucet stacks 75m STX
+    let response = await broadcastStackSTX(
+      2,
+      network,
+      75_000_000_000_000,
+      Accounts.FAUCET,
+      blockHeight,
+      cycles,
+      fee,
+      0
+    );
+    expect(response.error).toBeUndefined();
+
+    // Faucet delegates 1_100m to Bob
+    response = await broadcastDelegateSTX(
+      2,
+      network,
+      Accounts.FAUCET,
+      fee,
+      1,
+      1_100_000_000_000_000,
+      Accounts.WALLET_2
+    );
+    expect(response.error).toBeUndefined();
+    let [block, tx] = await waitForStacksTransaction(
+      orchestrator,
+      response.txid
+    );
+    expect(tx.success).toBeTruthy();
+
+    // Bob increase delegation by 920m
+    response = await broadcastDelegateStackIncrease(
+      2,
+      network,
+      Accounts.WALLET_2,
+      fee,
+      0,
+      Accounts.FAUCET,
+      Accounts.WALLET_2,
+      920_000_000_000_000
+    );
+    expect(response.error).toBeUndefined();
+
+    [block, tx] = await waitForStacksTransaction(orchestrator, response.txid);
+    expect(tx.success, tx.result).toBeTruthy();
+
+    // Bob commits 920m
```

Comment:
what the difference between committing STX and stacking STX?

Reply:
Stacking stx is used by direct stackers. We can also say that pooled stacker are stacking STX as part of a pool.

Committing STX is an action of the pool operator. The pool operator locks STX for their pool members, these are registered as partial stacked STX (map `partial-stacked-by-cycle`). Only when the pool operator calls `stack-aggregation-commit-indexed` these partial stacked STX will start to earn rewards.

If you want to read more: 
* How to run a stacking pool https://app.sigle.io/friedger.id/UOvy85BCSD-bjlrv_6q74
* Some FAQ (will add your question later): https://app.sigle.io/friedger.id/MeBQ49LDcr_ZD_afEwOwH
* Some Glossary for pooled stacking: https://app.sigle.io/friedger.id/MtU8D4thkohi5TVKBKIy_

Reply:
ok perfect